Cargo.toml

```diff
@@ -28,7 +28,7 @@ anyhow = "1.0.34"
 atty = "0.2.11"
 camino = "1.0.3"
 cargo_metadata = "0.14"
-clap = { version = "3.1", features = ["derive"] }
+clap = { version = "~3.1", features = ["derive"] } # https://github.com/clap-rs/clap/issues/3822
```

Comment:
This can now be reverted as deprecations are now off by default, see clap-rs/clap#3830 for details.  When you do start working through the deprecations as part of upgrading to 4.0, we've also improved the warnings for derive users.  See clap-rs/clap#3832 for examples.

btw is there a reason this project doesn't check-in the `Cargo.lock`?  Generally, [it is recommended to do so for programs](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries) and that would have prevented the unintended upgrade rather than needing to pin the dependency.

Reply:
> This can now be reverted as deprecations are now off by default, see [clap-rs/clap#3830](https://github.com/clap-rs/clap/pull/3830) for details. When you do start working through the deprecations as part of upgrading to 4.0, we've also improved the warnings for derive users. See [clap-rs/clap#3832](https://github.com/clap-rs/clap/pull/3832) for examples.

Nice! Thanks for working on this problem!

> btw is there a reason this project doesn't check-in the `Cargo.lock`? Generally, [it is recommended to do so for programs](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries) and that would have prevented the unintended upgrade rather than needing to pin the dependency.

This is intentional. See https://github.com/taiki-e/cargo-llvm-cov/pull/152#issuecomment-1107055622 for details.